p2.py

```diff
@@ -2,9 +2,13 @@
 #clink the above link to the problem description
 class Solution:
     def isPalindrome(self, x: 'int') -> 'bool':
-        x=str(x)
-        j=len(x)//2
-        for i in range(j):
-            if x[int(i)]!=x[int(len(x)-1-i)]:
-                return False
-        return True
+        y,sum1=x,0
+        if x<0:return False
+        else:
+            while y!=0:
+                dig=y%10
+                sum1=sum1*10+dig
+                y=y//10
+        if sum1==x:
+            return True
+        else:return False
```

Comment:
You don't have to do these changes this time. But it's good to keep these things in mind when coding. It results in elegant code.